### PR TITLE
Document syntactic restriction for ids in ledger-api proto

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
We document the syntactic restriction about string identifiers in the ledger api protobuf file

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/command_completion_service.proto [hunk 1]
```diff
@@ -42,14 +42,17 @@ service CommandCompletionService {
...
   // Required
   string ledger_id = 1;
 
   // Only completions of commands submitted with the same application_id will be visible in the stream.
+  // Must be a valid LedgerString (as described in ``value.proto``).
   // Required
   string application_id = 2;
 
   // Non-empty list of parties whose data should be included.
+  // Must be a valid PartyString.
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/command_completion_service.proto]:
```suggestion
  // Must be a valid PartyString (as described in ``value.proto``).
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/event.proto [hunk 2]
```diff
@@ -141,11 +153,16 @@ message ExercisedEvent {
...
   //
+  // Each element must be a valid PartyIdString (as described in ``proto.value``).
+  //
   // Required
   repeated string witness_parties = 10;
 
   // References to further events in the same transaction that appeared as a result of this ``ExercisedEvent``.
   // It contains only the immediate children of this event, not all members of the subtree rooted at this node.
+  //
+  // Each element must be a valid PartyString.
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/event.proto]:
```suggestion
  // Each element must be a valid LedgerString (as described in ``value.proto``).
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/event.proto [hunk 3]
```diff
@@ -141,11 +153,16 @@ message ExercisedEvent {
   // `choice ... controller` syntax, and said controllers are not
   // explicitly marked as observers.
   //
+  // Each element must be a valid PartyIdString (as described in ``proto.value``).
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/event.proto]:
```suggestion
  // Each element must be a valid PartyIdString (as described in ``value.proto``).
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/event.proto [hunk 4]
```diff
@@ -119,6 +129,8 @@ message ExercisedEvent {
   Value choice_argument = 6;
 
   // The parties that made the choice.
+  // Each element must be a valid PartyIdString (as described in
+  // ``proto.value``).
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/event.proto]:
```suggestion
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/event.proto [hunk 5]
```diff
@@ -119,6 +129,8 @@ message ExercisedEvent {
   Value choice_argument = 6;
 
   // The parties that made the choice.
+  // Each element must be a valid PartyIdString (as described in
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/event.proto]:
```suggestion
  // Each element must be a valid PartyIdString (as described in ``value.proto``).
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/package_service.proto [hunk 6]
```diff
@@ -40,17 +41,20 @@ message ListPackagesRequest {
 message ListPackagesResponse {
 
   // The IDs of all DAML-LF packages supported by the server.
+  // Each element must be a valid PackageIdString.
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/package_service.proto]:
```suggestion
  // Each element must be a valid PackageIdString (as described in ``value.proto``).
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/testing/reset_service.proto [hunk 7]
```diff
@@ -43,6 +43,7 @@ service ResetService {
 
 message ResetRequest {
   // Must correspond to the ledger ID reported by the Ledger Identification Service.
+  // Must be a valid ``LedgerString`` (as describe in ``value.proto``).
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/testing/reset_service.proto]:
```suggestion
  // Must be a valid LedgerString (as describe in ``value.proto``).
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/testing/time_service.proto [hunk 8]
```diff
@@ -28,6 +28,7 @@ service TimeService {
 message GetTimeRequest {
 
   // Must correspond to the ledger ID reported by the Ledger Identification Service.
+  // Must be a valid ``LedgerString`` (as describe in ``value.proto``).
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/testing/time_service.proto]:
```suggestion
  // Must be a valid LedgerString (as describe in ``value.proto``).
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/testing/time_service.proto [hunk 9]
```diff
@@ -41,6 +42,7 @@ message GetTimeResponse {
 message SetTimeRequest {
 
   // Must correspond to the ledger ID reported by the Ledger Identification Service.
+  // Must be a valid ``LedgerString`` (as describe in ``value.proto``).
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/testing/time_service.proto]:
```suggestion
  // Must be a valid LedgerString (as describe in ``value.proto``).
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/transaction.proto [hunk 10]
```diff
@@ -40,10 +43,12 @@ message TransactionTree {
   string offset = 6;
 
   // Changes to the ledger that were caused by this transaction. Nodes of the transaction tree.
+  // Each key be a valid ``LedgerString`` (as describe in ``value.proto``).
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/transaction.proto]:
```suggestion
  // Each key must be a valid LedgerString (as describe in ``value.proto``).
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/transaction.proto [hunk 11]
```diff
@@ -40,10 +43,12 @@ message TransactionTree {
...
   // Required
   map<string, TreeEvent> events_by_id = 7;
 
   // Roots of the transaction tree.
+  // Each element be a valid ``LedgerString`` (as describe in ``value.proto``).
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/transaction.proto]:
```suggestion
  // Each element must be a valid LedgerString (as describe in ``value.proto``).
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/transaction_service.proto [hunk 12]
```diff
@@ -109,15 +113,18 @@ message GetTransactionByEventIdRequest {
 
 message GetTransactionByIdRequest {
   // Must correspond to the ledger ID reported by the Ledger Identification Service.
+  // Must be a valid ``LedgerString`` (as describe in ``value.proto``).
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/transaction_service.proto]:
```suggestion
  // Must be a valid LedgerString (as describe in ``value.proto``).
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/transaction_service.proto [hunk 13]
```diff
@@ -109,15 +113,18 @@ message GetTransactionByEventIdRequest {
...
   // Required
   string ledger_id = 1;
 
   // The ID of a particular transaction.
+  // Must be a valid ``LedgerString`` (as describe in ``value.proto``).
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/transaction_service.proto]:
```suggestion
  // Must be a valid LedgerString (as describe in ``value.proto``).
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/transaction_service.proto [hunk 14]
```diff
@@ -109,15 +113,18 @@ message GetTransactionByEventIdRequest {
...
   // Required
   string transaction_id = 2;
 
   // The parties whose events the client expects to see.
   // Events that are not visible for the parties in this collection will not be present in the response.
+  // Each element be a valid ``PartyIdString`` (as describe in ``value.proto``).
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/transaction_service.proto]:
```suggestion
  // Each element be a valid PartyIdString (as describe in ``value.proto``).
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/transaction_service.proto [hunk 15]
```diff
@@ -137,6 +144,7 @@ message GetFlatTransactionResponse {
 
 message GetLedgerEndRequest {
   // Must correspond to the ledger ID reported by the Ledger Identification Service.
+  // Must be a valid ``LedgerString`` (as describe in ``value.proto``).
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/transaction_service.proto]:
```suggestion
  // Must be a valid LedgerString (as describe in ``value.proto``).
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/value.proto [hunk 16]
```diff
@@ -101,6 +112,7 @@ message RecordField {
 message Identifier {
 
   // The identifier of the DAML package that contains the entity.
+  // Must be a valid `PackageIdString`.
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/value.proto]:
```suggestion
  // Must be a valid PackageIdString.
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/value.proto [hunk 17]
```diff
@@ -90,6 +100,7 @@ message RecordField {
   // When submitting a commmand, it's optional:
   //   - if all keys within a single record are present, the order in which fields appear does not matter. however, each key must appear exactly once.
   //   - if any of the keys within a single record are omitted, the order of fields MUST match the order of declaration in the DAML template.
+  // Must be a valid ``NameString``
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/value.proto]:
```suggestion
  // Must be a valid NameString
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/value.proto [hunk 18]
```diff
@@ -126,6 +138,7 @@ message Variant {
   Identifier variant_id = 1;
 
   // Determines which of the Variant's alternatives is encoded in this message.
+  // Must be a valid ``NameString``.
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/value.proto]:
```suggestion
  // Must be a valid NameString.
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/value.proto [hunk 19]
```diff
@@ -142,6 +155,7 @@ message Enum {
   Identifier enum_id = 1;
 
   // Determines which of the Variant's alternatives is encoded in this message.
+  // Must be a valid ``NameString``.
```
Comment (gerolf-da) [inline ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/value.proto]:
```suggestion
  // Must be a valid NameString.
```

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/command_completion_service.proto (hunk 1 above)
Comment (remyhaemmerle-da) [reply ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/command_completion_service.proto]: fixed

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/event.proto (hunk 3 above)
Comment (remyhaemmerle-da) [reply ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/event.proto]: fixed

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/event.proto (hunk 2 above)
Comment (remyhaemmerle-da) [reply ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/event.proto]: fixed

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/event.proto (hunk 4 above)
Comment (remyhaemmerle-da) [reply ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/event.proto]: fixed

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/event.proto (hunk 5 above)
Comment (remyhaemmerle-da) [reply ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/event.proto]: fixed

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/value.proto (hunk 18 above)
Comment (remyhaemmerle-da) [reply ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/value.proto]: fixed

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/value.proto (hunk 17 above)
Comment (remyhaemmerle-da) [reply ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/value.proto]: fixed

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/value.proto (hunk 19 above)
Comment (remyhaemmerle-da) [reply ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/value.proto]: fixed

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/transaction_service.proto (hunk 15 above)
Comment (remyhaemmerle-da) [reply ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/transaction_service.proto]: fixed

File: ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/value.proto (hunk 16 above)
Comment (remyhaemmerle-da) [reply ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/value.proto]: fixed

**Pull request merged**